### PR TITLE
migrate to sops-nix

### PR DESCRIPTION
**Pull request by Mic92**
This gets rid of slow nixops key deployment and git-crypt in long-term

Would you like to use sops-nix with pgp or age?
cc @nlewo @adisbladis @flokli @ryantm @zimbatm @timokau

### CODE REVIEW

File: .sops.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+keys:
+  - &mic92 age17n64ahe3wesh8l8lj0zylf4nljdmqn28hvqns2g7hgm9mdkhlsvsjuvkxz
```
Comment (Mic92) [inline .sops.yaml:2]: This can be either a pgp fingerprint or an age key.

Comment (Mic92) [reply .sops.yaml:2]: For gpg: https://github.com/Mic92/sops-nix#2a-generate-a-gpg-key-for-yourself
For age: https://github.com/Mic92/sops-nix#2a-generate-a-ssh-and-age-key-for-yourself

**Comment by nlewo** [comment]
I never used age. @Mic92 Is it possible to store age key in a yubikey?

**Comment by Mic92** [comment]
> I never used age. @Mic92 Is it possible to store age key in a yubikey?

If you want to use your yubikey than gnupg is still the better choice. I heard newer yubikeys also work with ed25519 (the crypto algorithms used by age) https://github.com/str4d/age-plugin-yubikey 
But I don't think this has been integrated with sops.

File: .sops.yaml (hunk 1 above)
Comment (ryantm) [reply .sops.yaml:2]: Oh, is there not support for using SSH keys directly?

Comment (Mic92) [reply .sops.yaml:2]: No the upstream Age maintainer did not want this when reviewing the sops PR that added age support. However we build an easy to use tool that does the conversion.

**Comment by Mic92** [comment]
Here is an example what adding a gpg key looks like: https://github.com/Mic92/dotfiles/blob/f94773404d6ded8ac4e338f8cbaa2d2b5b914e57/nixos/.sops.yaml#L2

**Comment by zimbatm** [comment]
Can we stick to GPG for now since we all already have a key for it? That should allow making the migration transparent.

**Comment by Mic92** [comment]
> Can we stick to GPG for now since we all already have a key for it? That should allow making the migration transparent.
> 
> Age looks cool but I need to look into how to backup/restore the private key properly.

Sure everyone can use the key format they prefer.

**Comment by Mic92** [comment]
Please share your pgp or age key fingerprint here. I tried to use the git-crypt one, but gpg says those are invalid for most users.

**Comment by nlewo** [comment]
@Mic92 my pgp key fingerprint is `B31F04575C4561F58DE02C9EFC6C2E50B00E9861`.

**Comment by Mic92** [comment]
> @Mic92 my pgp key fingerprint is `B31F04575C4561F58DE02C9EFC6C2E50B00E9861`.

Please upload your public key to https://github.com/nix-community/infra/tree/master/keys
I could not find your key on either ubuntu or sks key server.

**Pull request merged**